src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java

```diff
@@ -22,8 +26,29 @@ public ZipFileStatusService(ProcessEventRepository eventRepo, EnvelopeRepository
 
     public ZipFileStatus getStatusFor(String zipFileName) {
         return new ZipFileStatus(
-            envelopeRepo.findByZipFileName(zipFileName),
-            eventRepo.findByZipFileName(zipFileName)
+            envelopeRepo
+                .findByZipFileName(zipFileName)
+                .stream()
+                .map(envelope ->
+                    new ZipFileEnvelope(
+                        envelope.getId().toString(),
+                        envelope.getContainer(),
+                        envelope.getStatus().name()
+                    )
+                )
+                .collect(toList()),
+            eventRepo
+                .findByZipFileName(zipFileName)
+                .stream()
+                .map(event ->
+                    new ZipFileEvent(
+                        event.getEvent().name(),
+                        event.getContainer(),
+                        event.getCreatedAt()
+                    )
+                )
+                .collect(toList())
+
```

Comment:
instead of evaluating the into the list and re-streaming again we could make repository return streams instead. so code would be similar to following:

```java
try (Stream stream = repo.findByCondition(value)) {
    return stream.map(...).collect(toList());
}
```